src/middlewares/jobsMiddleware.js

```diff
@@ -0,0 +1,68 @@
+import find from 'lodash.find';
+import isUndefined from 'lodash.isundefined';
+
+
+const JOBS = [
+  {
+    create: { controllerId: 'AnalysisController', operationId: 'createUrlExport' },
+    poll: { controllerId: 'AnalysisController', operationId: 'getUrlExportStatus', jobIdKey: 'urlExportId' },
+  },
+  {
+    create: { controllerId: 'AnalysisController', operationId: 'createPDFExport' },
+    poll: { controllerId: 'AnalysisController', operationId: 'getPdfExportStatus', jobIdKey: 'pdfExportId' },
+  },
+  {
+    create: { controllerId: 'SegmentController', operationId: 'createDryRun' },
+    poll: { controllerId: 'SegmentController', operationId: 'getDryRunSegmentStatus', jobIdKey: 'dryRunId' },
+  },
+];
+
+export default function({
+  jobs = JOBS,
+  pollInterval = 5000,
+} = {}) {
+  return function jobsMiddleware({ controllerId, operationId, baseSdk }) {
+    return next => function(params, callback, options) {
```

Comment:
anonymous function should expect, arrow and name function both have their use case